doc/flame/components.md

```diff
@@ -419,19 +419,50 @@ use `animation.completed`.
 Example:
 
 ```dart
-await animation.completed;
+await
+animation.completed;
+
 doSomething();
 
 // or alternatively
 
 animation.completed.whenComplete(doSomething);
 ```
 
+Additionally, this component also has the following optional event callbacks:  `onStart`, `onFrame`,
+and `onComplete`. To listen to these events, you can do the following:
+
+```dart
+
+final animation =
+SpriteAnimation.spriteList([sprite], stepTime: 1, loop: false)
+  ..onStart = () =>
+  {
+    //do something};
+  };
+
+final animation =
+SpriteAnimation.spriteList([sprite], stepTime: 1, loop: false)
+  ..onComplete = () =>
+  {
+    //do something};
+  };
+
+final animation =
+SpriteAnimation.spriteList(spriteList, stepTime: 1, loop: false);
+animation.onFrame = (
+index) {
+if(index===1){
+//do something for the second frame
+}
+};
+```
 
 ## SpriteAnimationGroup
 
 `SpriteAnimationGroupComponent` is a simple wrapper around `SpriteAnimationComponent` which enables
-your component to hold several animations and change the current playing animation in runtime.
+your component to hold several animations and change the current playing animation in runtime. As
+this component is just a wrapper, the event listeners can also be implemented as detailed above.
```

Comment:
```suggestion
your component to hold several animations and change the current playing animation at runtime. Since
this component is just a wrapper, the event listeners can be implemented as described in
[](#spriteanimationcomponent).
```